Install PyPDF2 for reading the PDF File.

In [ ]:
!pip install PyPDF2

In [ ]:
#Import the libraries
import google.generativeai as genai
import PyPDF2 as pdf
import json

Create a API Key using Google Studio.

In [ ]:
genai.configure(api_key = "Your API Key")

Use google.generativeai for taking our response from Google Gemini and use PyPDF2 for read our PDF file.

In [ ]:
def get_gemini_repsonse(input):
    model=genai.GenerativeModel('gemini-pro')
    response=model.generate_content(input)
    return response.text

def input_pdf_text(uploaded_file):
    reader=pdf.PdfReader(uploaded_file)
    text=""
    for page in range(len(reader.pages)):
        page=reader.pages[page]
        text+=str(page.extract_text())
    return text

Check the Gemini response work properly

In [ ]:
get_gemini_repsonse("Hi")

'Hello. How can I help?'

Build a prompt for the gemini. So that it will give a accurate Summary (Output) according to the data which is provided in the PDF.

In [ ]:
def build_prompt(text):
    input_prompt=f"""
    As the Chief Financial Officer (CFO) of a company,
    your role is to present a summary of the company's financial
    results for the following details.
    Provide an overview that includes key metrics such as revenue,
    net income, operating expenses, and cash flow.
    Highlight significant achievements or setbacks,
    compare the results with the previous period,
    and identify any trends or factors that influenced the
    performance. Your summary should be clear, data-driven,
    and provide insights for stakeholders to understand the
    company's financial health and strategic direction
    Results Document:{text}

    Provide the summary as a list.
    I want the response in one single string having the structure
    {{"Summary:[]"}}
    """
    return input_prompt

Upload PDF file in the below code.

In [ ]:
text=input_pdf_text("/content/hdfc.pdf")

Input text in the prompt.

In [ ]:
input_prompt = build_prompt(text)

Stored Gemini Response in response

In [ ]:
response=get_gemini_repsonse(input_prompt)
response = json.loads(response)

Import textwarp for beautification of output result. Use new lines and other bullets for asthetic purpose.

In [ ]:
import textwrap


def add_newlines(text):
  sentences = text.split('. ')
  return '.\n'.join(sentences) + '.\n'

def print_response(response):
  print("Summary:")
  for suggestion in response["Summary"]:
      print("- {}".format(add_newlines(suggestion)))

print_response(response)

Summary:
- 1.
Revenue saw a significant increase of 15% compared to the previous period, driven by strong sales in our core product lines and the launch of new products..

- 2.
Net income grew by an impressive 20%, reflecting our ongoing cost optimization efforts and improved operational efficiency..

- 3.
Operating expenses remained largely stable, indicating our commitment to prudent financial management..

- 4.
Cash flow from operations showed a notable improvement, providing us with increased financial flexibility and stability..

- 5.
Our financial performance exceeded industry benchmarks, showcasing the strength of our business model and our ability to navigate challenging economic conditions..



#Check PDF Response - File HDFC

#**Summary:**
- 1.
Revenue saw a significant increase of 15% compared to the previous period, driven by strong sales in our core product lines and the launch of new products..

- 2.
Net income grew by an impressive 20%, reflecting our ongoing cost optimization efforts and improved operational efficiency..

- 3.
Operating expenses remained largely stable, indicating our commitment to prudent financial management..

- 4.
Cash flow from operations showed a notable improvement, providing us with increased financial flexibility and stability..

- 5.
Our financial performance exceeded industry benchmarks, showcasing the strength of our business model and our ability to navigate challenging economic conditions..


#Check Other PDF Response - File HDFC Result

Summary:
- **Overview:** HDFC Bank Ltd.
presented strong financial performance in Q1 FY 2024-25 with key metrics showing positive growth.

**Key Metrics:**
* Net Interest Income increased by 26.4% to ₹298.4 billion from ₹236.0 billion in Q1 of the previous year.
* Total advances grew by 24.4% to ₹23,791 billion from ₹19,131 billion.
* Total deposits increased by 35.3% to ₹25,017 billion from ₹18,507 billion.
* Net profit rose by 52.5% to ₹119.5 billion from ₹78.3 billion.

**Significant Achievements:**
* Net interest margin improved to 3.7%, contributing to higher net interest income.
* Retail advances grew by 100.4%, driven by strong demand for mortgages and personal loans.
* CASA ratio increased to 36%, reflecting a shift towards lower-cost deposits.
* Fee and commission income grew by 12.0% to ₹62.9 billion.

**Challenges:**
* Gross NPA ratio increased slightly to 1.33% from 1.23% in the previous quarter, reflecting the impact of the economic slowdown.

**Outlook:** HDFC Bank remains optimistic about the future, focusing on increasing its market share in retail and corporate banking while managing risks prudently..